diff --git a/src/network/requestQueue/index.js b/src/network/requestQueue/index.js
index eee14d66c3869d90f7bac0ac755ecbad3f2a2741..5fa67064e471d296b7c7992d8aded5ef7181d6b7 100644
--- a/src/network/requestQueue/index.js
+++ b/src/network/requestQueue/index.js
@@ -309,7 +309,7 @@ module.exports = class RequestQueue extends EventEmitter {
     // will be fine, and potentially fix up a new timeout if needed at that time.
     // Note that if we're merely "overloaded" by having too many inflight requests
     // we will anyways check the queue when one of them gets fulfilled.
-    let scheduleAt = this.throttledUntil - Date.now()
+    let scheduleAt = Math.max(0, this.throttledUntil - Date.now())
     if (!this.throttleCheckTimeoutId) {
       if (this.pending.length > 0) {
         scheduleAt = scheduleAt > 0 ? scheduleAt : CHECK_PENDING_REQUESTS_INTERVAL
